src/libcollections/string.rs

```diff
@@ -1129,8 +1129,6 @@ impl String {
     #[inline]
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn insert(&mut self, idx: usize, ch: char) {
-        let len = self.len();
-        assert!(idx <= len);
```

Comment:
It's redundant but it also affects the panic the users sees. The one below is the same, the last one is truly redundant.